### PR TITLE
Ria 2573 ftpa decision resident judge

### PR DESCRIPTION
**Pull request by burale**
### JIRA link (if applicable) ###

[RIA-2573](https://tools.hmcts.net/jira/browse/RIA-2573)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/resources/scenarios/RIA-2573-ftpa-appeal-decision-allowed-resident-judge.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        ]
+      }
+    },
+    "notifications": [
+      {
+        "reference": "1002_FTPA_APPLICATION_DECISION_HOME_OFFICE",
+        "recipient": "{$respondentEmailAddresses.respondentEvidenceDirection}",
+        "subject": "Immigration and Asylum appeal: application to appeal to Upper Tribunal",
+        "body": [
+          "HMCTS reference: PA/12345/2019",
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-2573-ftpa-appeal-decision-allowed-resident-judge.json:45]: You need to remove static content "HMCTS reference:" (and similar) from these functional tests.

Comment (burale) [reply src/functionalTest/resources/scenarios/RIA-2573-ftpa-appeal-decision-allowed-resident-judge.json]: Fixed

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/entities/ApplicantType.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    RESPONDENT("respondent");
+
+    @JsonValue
+    private final String value;
+
+    ApplicantType(String value) {
+        this.value = value;
+    }
+
+    public static Optional<ApplicantType> from(
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/entities/ApplicantType.java:19]: May be better to have a more descriptive method name here.

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java [hunk 3]
```diff
@@ -623,20 +623,22 @@ private boolean isValidUserDirection(
                 boolean isNegative = asylumCase
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:623]: variable 'isNegative' doesn't make too much sense to me. Not sure if we could have a more appropriate name to better describe this boolean?

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java [hunk 4]
```diff
@@ -649,10 +651,99 @@ private boolean isValidUserDirection(
                     .orElse(false);
 
                 return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-                       && callback.getEvent() == Event.FTPA_DECISION_LEADERSHIP_JUDGE
+                       && (callback.getEvent() == Event.LEADERSHIP_JUDGE_FTPA_DECISION || callback.getEvent() == Event.RESIDENT_JUDGE_FTPA_DECISION)
                        && isPositive;
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:655]: Similar to above. I would be in favour of a variable named 'isPostiveOutcome' instead.

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java [hunk 5]
```diff
@@ -649,10 +651,99 @@ private boolean isValidUserDirection(
...
+
+    @Bean
+    public PreSubmitCallbackHandler<AsylumCase> ftpaApplicationDecisionReheardAppellantNotificationHandler(
+        @Qualifier("ftpaApplicationDecisionReheardAppellantNotificationGenerator") List<NotificationGenerator> notificationGenerators) {
+
+        return new NotificationHandler(
+            (callbackStage, callback) -> {
+
+                AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+                boolean isPositive = asylumCase
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:669]: Similar to above. I would be in favour of a variable named 'isPostiveOutcome' instead.

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -209,6 +209,13 @@ govnotify:
     applicationRefusedApplicant: ab6b3c49-7f7e-407a-b46d-8dda10e33803
     applicationRefusedOtherParty: 59ec29bf-5d40-4a97-bd54-cf7262b65185
 
+    applicationReheardApplicant: ac94290a-bc6a-4e3b-a9cb-34d48400450c
```
Comment (bradsorour) [inline src/main/resources/application.yaml:212]: I think we need to apply same formatting as the other notification templates here. For example:

> applicationReheard:
>     applicant: <template>
>     otherParty: <template>

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java (hunk 5 above)
Comment (burale) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:669]: Changed

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java (hunk 3 above)
Comment (burale) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java:623]: Changed

File: src/main/resources/application.yaml (hunk 6 above)
Comment (burale) [reply src/main/resources/application.yaml:212]: This has to go through at the moment. I am refactoring it on RIA-2975

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/adminofficer/AdminOfficerFtpaDecisionPersonalisation.java [hunk 7]
```diff
@@ -51,9 +54,18 @@ public String getReferenceId(Long caseId) {
...
+        Optional<FtpaDecisionOutcomeType> ftpaDecisionOutcomeType = asylumCase
+            .read(FTPA_APPELLANT_DECISION_OUTCOME_TYPE, FtpaDecisionOutcomeType.class);
+
+        if (!ftpaDecisionOutcomeType.isPresent()) {
+            ftpaDecisionOutcomeType = asylumCase
+                .read(FTPA_RESPONDENT_DECISION_OUTCOME_TYPE, FtpaDecisionOutcomeType.class);
+        }
+        return ftpaDecisionOutcomeType.isPresent()
+            ? ftpaDecisionOutcomeType.get()
+            : null;
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/adminofficer/AdminOfficerFtpaDecisionPersonalisation.java:69]: Not sure if you can reduce the number of read calls by using the following:

```
return asylumCase.read(FTPA_APPELLANT_DECISION_OUTCOME_TYPE, FtpaDecisionOutcomeType.class).isPresent() 
    	? asylumCase.read(FTPA_APPELLANT_DECISION_OUTCOME_TYPE, FtpaDecisionOutcomeType.class).get() 
    	: null
```

**Comment by makowalski** [comment]
changes are released within RIA-2975

**Pull request closed without merging**